core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -505,7 +525,29 @@ public void refresh(String source) throws EngineException {
             throw new RefreshFailedEngineException(shardId, t);
         }
 
-        // TODO: maybe we should just put a scheduled job in threadPool?
+        mergeScheduler.refreshConfig();
```

Comment:
it's done in refreshInternal as well.. so I'm probably missing something w.r.t. lucene.  Looking to learn ;)